packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -44,11 +46,16 @@ class Routes extends React.Component {
 
     render() {
         const { error, has_error, is_logged_in, passthrough } = this.props;
+        const lang = getLanguage();
 
         if (has_error) {
             return <Error {...error} />;
         }
 
+        if (!/[?&]lang=/.test(location.search) && lang !== 'EN') {
+            window.history.replaceState({}, document.title, urlForLanguage(lang));
```

Comment:
why do we need to preserve language params on every route change?, if its intentional, pls add comment here

Reply:
Ok, added. Please, check if it's sufficient.